### ISSUE TITLE
Profiler: Data race. 

### ISSUE THREAD

**Issue by hackintoshrao** [minio/minio#1617]
Here is how to reproduce 
- Just buiild with -race flag. `go build -race` . 
- Set MINIO_PROFILER 
- Start the server and press Ctrl + C. 

The following the server console log 

```
Minio Browser:
    http://127.0.0.1:9001
    http://172.31.10.105:9001

To configure Minio Client:
    $ wget https://dl.minio.io/client/mc/release/linux-amd64/mc
    $ chmod 755 mc
    $ ./mc config host add myminio http://127.0.0.1:9001 A2Q0GSBFSVH6UUEKYLFW IurnSMkeSnzzCGYjVr5qpa5OH/9Bh9ZCbWihZ/f3
^C2016/05/12 14:44:30 profile: caught interrupt, stopping profiles
==================
WARNING: DATA RACE
Read by goroutine 9:
  github.com/minio/minio/vendor/github.com/pkg/profile.(*profile).Stop()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:100 +0x79
  github.com/minio/minio/vendor/github.com/pkg/profile.Start.func5()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:188 +0x20c

Previous write by main goroutine:
  [failed to restore the stack]

Goroutine 9 (running) created at:
  github.com/minio/minio/vendor/github.com/pkg/profile.Start()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:191 +0xa14
  main.main()
      /home/ec2-user/mygo/src/github.com/minio/minio/main.go:191 +0x1b7
==================
==================
WARNING: DATA RACE
Read by goroutine 9:
  github.com/minio/minio/vendor/github.com/pkg/profile.(*profile).Stop()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:100 +0xc6
  github.com/minio/minio/vendor/github.com/pkg/profile.Start.func5()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:188 +0x20c

Previous write by main goroutine:
  [failed to restore the stack]

Goroutine 9 (running) created at:
  github.com/minio/minio/vendor/github.com/pkg/profile.Start()
      /home/ec2-user/mygo/src/github.com/minio/minio/vendor/github.com/pkg/profile/profile.go:191 +0xa14
  main.main()
      /home/ec2-user/mygo/src/github.com/minio/minio/main.go:191 +0x1b7
==================
Found 2 data race(s)

```

Profiler fails with prof file too. 
`$go tool pprof ~/.minio/profile/cpu.pprof /mnt/rao/minio`
`parsing profile: unrecognized profile format`

**Comment by hackintoshrao** [comment]
Using devcheney/profile currently and getting it to run meanwhile.

**Comment by harshavardhana** [comment]
The `pkg/profile` is from devcheney, its a newer package than older one at devcheney/profile.  We should fix this for davecheney's `pkg/profile` package. 

But not holding this as priority for the release.

**Comment by harshavardhana** [comment]
Assigned to you - so that you can send a PR to the relevant package `pkg/profile` ?

**Comment by hackintoshrao** [comment]
Sure 👍

**Comment by davecheney** [comment]
This should be fixed by pkg/profile@c78aac22bd43883fd2817833b982153dcac17b3b

**Comment by harshavardhana** [comment]
Fixed by @davecheney upstream.
